brainglobe_stitch/image_mosaic.py

```diff
@@ -340,3 +346,366 @@ def read_big_stitcher_transforms(self) -> None:
+        downscale_factors: Tuple[int, int, int] = (1, 2, 2),
+    ) -> None:
+        """
+        Fuse the tiles into a single image and save it to the output file.
+
+        Parameters
+        ----------
+        output_file_name: str, default: "fused.zarr"
+            The name of the output file, suffix dictates the output file type.
+        downscale_factors: Tuple[int, int, int], default: (1, 2, 2)
+            The factors to downscale the image by in the z, y, x dimensions.
+        """
+        output_path = self.directory / output_file_name
+
+        z_size, y_size, x_size = self.tiles[0].data_pyramid[0].shape
+        # Calculate the shape of the fused image
+        fused_image_shape: Tuple[int, ...] = (
+            max([tile.position[0] for tile in self.tiles]) + z_size,
+            max([tile.position[1] for tile in self.tiles]) + y_size,
+            max([tile.position[2] for tile in self.tiles]) + x_size,
+        )
+
+        if output_path.suffix == ".zarr":
+            self._fuse_to_zarr(
+                output_path,
+                fused_image_shape,
+                downscale_factors=downscale_factors,
+            )
+        elif output_path.suffix == ".h5":
+            self._fuse_to_bdv_h5(
+                output_path,
+                fused_image_shape,
+                downscale_factors=downscale_factors,
+            )
+
+    def _fuse_to_zarr(
+        self,
+        output_path: Path,
+        fused_image_shape: Tuple[int, ...],
+        pyramid_depth: int = 5,
+        chunk_shape: Tuple[int, ...] = (128, 128, 128),
+        compression: str = "zstd",
+        compression_level: int = 3,
+        downscale_factors: Tuple[int, int, int] = (1, 2, 2),
+    ) -> None:
+        """
+        Fuse the tiles in the ImageMosaic into a single image and save it as a
+        zarr file.
+
+        Parameters
+        ----------
+        output_path: Path
+            The path of the output file.
+        fused_image_shape: Tuple[int, ...]
+            The shape of the fused image.
+        pyramid_depth: int, default: 5
+            The depth of the resolution pyramid.
+        chunk_shape: Tuple[int, ...], default: (128, 128, 128)
+            The shape of the chunks in the zarr file.
+        compression: str, default: "zstd"
+            The compression algorithm to use.
+        compression_level: int, default: 3
+            The compression level to use.
+        downscale_factors: Tuple[int, int, int], default: (1, 2, 2)
+            The factors to downscale the image by in the z, y, x dimensions.
+        """
+        z_size, y_size, x_size = self.tiles[0].data_pyramid[0].shape
+
+        transformation_metadata, axes_metadata = self.get_metadata_for_zarr(
+            pyramid_depth=pyramid_depth, downscale_factors=downscale_factors
+        )
+
+        if self.num_channels > 1:
+            fused_image_shape = (self.num_channels, *fused_image_shape)
+            chunk_shape = (self.num_channels, *chunk_shape)
```

Comment:
(I dont know very much about zarr)

Reply:
The chunk_shape defines how the data is stored under the hood (in case of `zarr` the shape of each individual file). From what I understand, this is the minimum unit that is moved between RAM and disk when the files are loaded for analysis and visualisation. I had to strike a balance between `chunk_shape` and number of files in case of zarr as each chunk is stored as a separate file. Initially I was working with a smaller `chunk_shape` (32, 32, 32) however, with large images this was generating 10k - 100k files.

An explanation similar to this will go into the documentation as this should be one of the parameters that's exposed to the user since for some, if they always view the resulting images in the same orientation something like (1, `image_shape[1]`, `image_shape[2]`) would make more sense.